seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/SeaTunnelSourceCollector.java

```diff
@@ -77,17 +87,26 @@ public SeaTunnelSourceCollector(
             List<OneInputFlowLifeCycle<Record<?>>> outputs,
             MetricsContext metricsContext,
             FlowControlStrategy flowControlStrategy,
-            SeaTunnelDataType rowType) {
+            SeaTunnelDataType rowType,
+            List<String> tableNames) {
         this.checkpointLock = checkpointLock;
         this.outputs = outputs;
         this.rowType = rowType;
+        this.metricsContext = metricsContext;
         if (rowType instanceof MultipleRowType) {
             ((MultipleRowType) rowType)
                     .iterator()
-                    .forEachRemaining(
-                            type -> {
-                                this.rowTypeMap.put(type.getKey(), type.getValue());
-                            });
+                    .forEachRemaining(type -> this.rowTypeMap.put(type.getKey(), type.getValue()));
+        }
+        if (CollectionUtils.isNotEmpty(tableNames)) {
+            tableNames.forEach(
+                    tableName ->
+                            sourceReceivedCountPerTable.put(
+                                    TablePath.of(tableName).getTableName(),
```

Comment:
why not direct use `tableName`? It should contains more information like database name and schema name.